tests/integration/backup/self_reference/import_test.go

```diff
@@ -0,0 +1,303 @@
+		Actions: []any{
+			testUtils.SchemaUpdate{
+				Schema: `
+					type Author {
+						name: String
+						book: Book @relation(name: "author_book")
+						reviewed: Book @relation(name: "reviewedBy_reviewed")
+					}
+					type Book {
+						name: String
+						author: Author @primary @relation(name: "author_book")
+						reviewedBy: Author @primary @relation(name: "reviewedBy_reviewed")
+					}
+				`,
+			},
+			testUtils.BackupImport{
+				ImportContent: `{
+					"Author":[
+						{
+							"name":"John"
+						}
+					],
+					"Book":[
+						{
+							"name":"John and the sourcerers' stone",
+							"author":"bae-decf6467-4c7c-50d7-b09d-0a7097ef6bad",
+							"reviewedBy":"bae-decf6467-4c7c-50d7-b09d-0a7097ef6bad"
+						}
+					]
+				}`,
+			},
+			testUtils.Request{
+				Request: `
+					query {
+						Book {
+							name
+							author {
+								name
+								reviewed {
+									name
+								}
+							}
+						}
+					}`,
+				Results: []map[string]any{
+					{
+						"name": "John and the sourcerers' stone",
+						"author": map[string]any{
+							"name": "John",
+							"reviewed": map[string]any{
+								"name": "John and the sourcerers' stone",
+							},
+						},
+					},
+				},
+			},
+		},
+	}
+
+	testUtils.ExecuteTestCase(t, []string{"Author", "Book"}, test)
+}
+
+func TestBackupSelfRefImport_PrimaryRelationWithSecondCollectionWrongOrder_NoError(t *testing.T) {
+	test := testUtils.TestCase{
+		Actions: []any{
+			testUtils.SchemaUpdate{
+				Schema: `
+					type Author {
+						name: String
+						book: Book @relation(name: "author_book")
+						reviewed: Book @relation(name: "reviewedBy_reviewed")
+					}
+					type Book {
+						name: String
+						author: Author @primary @relation(name: "author_book")
+						reviewedBy: Author @primary @relation(name: "reviewedBy_reviewed")
+					}
+				`,
+			},
+			testUtils.BackupImport{
+				ImportContent: `{
+					"Book":[
+						{
+							"name":"John and the sourcerers' stone",
+							"author":"bae-decf6467-4c7c-50d7-b09d-0a7097ef6bad",
+							"reviewedBy":"bae-decf6467-4c7c-50d7-b09d-0a7097ef6bad"
+						}
+					],
+					"Author":[
+						{
+							"name":"John"
+						}
+					]
+				}`,
+			},
+			testUtils.Request{
+				Request: `
+					query {
+						Book {
+							name
+							author {
+								name
+								reviewed {
+									name
+								}
+							}
+						}
+					}`,
+				Results: []map[string]any{
+					{
+						"name": "John and the sourcerers' stone",
+						"author": map[string]any{
+							"name": "John",
+							"reviewed": map[string]any{
+								"name": "John and the sourcerers' stone",
+							},
+						},
+					},
+				},
+			},
+		},
+	}
+
+	testUtils.ExecuteTestCase(t, []string{"Author", "Book"}, test)
+}
+
+// This test documents undesirable behaviour, as the documents are not linked.
+// https://github.com/sourcenetwork/defradb/issues/1697
+func TestBackupSelfRefImport_SplitPrimaryRelationWithSecondCollection_NoError(t *testing.T) {
```

Comment:
suggestion: This test would work if the import content was taken from the generated export. I suggest either it documents an invalid import content and we remove the linked issue above the function name or we change the test to use a valid import content.

Reply:
That's fair, and it is more descriptive, even if a little longer/more-complex. Will change.

- [ ] import an export